istioctl/pkg/multixds/gather.go

```diff
@@ -128,8 +128,8 @@ func queryEachShard(all bool, dr *discovery.DiscoveryRequest, istioNamespace str
 		if err != nil {
 			return nil, err
 		}
-		defer fw.Close()
 		xdsOpts.Xds = fw.Address()
+		fw.Close()
```

Comment:
Why? Is this fixing any real problems?

Reply:
Sorry, I misunderstood for, chan and defer, there will be no leaks here